tests/test_private_attributes.py

```diff
@@ -174,11 +173,19 @@ def __init__(self):
 
     assert Model.__private_attributes__ == {}
     assert set(Model.__slots__) == {'foo', '_bar'}
-    m = Model()
+    m1 = Model()
+    m2 = Model()
+
     for attr in Model.__slots__:
-        assert object.__getattribute__(m, attr) == 'spam'
-        with pytest.raises(ValueError, match=f'"Model" object has no field "{attr}"'):
-            setattr(m, attr, 'not spam')
+        assert object.__getattribute__(m1, attr) == 'spam'
+
+    # In v2, you are always allowed to set instance attributes if the name starts with `_`.
```

Comment:
For reference: https://github.com/pydantic/pydantic/blob/0f3d477f32a76e37112a9d65d9a677da354f5317/pydantic/main.py#L230-L233

I'm assuming this was an intentional change — let me know if not